ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+                            [x_vals[dim_num]]), ndmin=2))
+                        for dim_num in range(space.n_dims) if dim_num not in (i,)]
+                    value_list, stddev_list = generate_random_data(
+                        locked_val_list=locked_val_list,
+                        sample_list=sample_list,
+                        model=model)
+                zi, stddevs = find_average_1D(
+                    value_list,
+                    stddev_list,
+                    included_list[i])
+                row.append({"xi": xi_list[i], "zi": zi, "std": stddevs})
+                # Finding the minimum and maximum objective value shown, accounting for
+                # uncertainty (if shown).
+                if show_confidence:
+                    zi_low_bound = zi - 1.96 * stddevs
+                    zi_high_bound = zi + 1.96 * stddevs
+                else:
+                    zi_low_bound = zi
+                    zi_high_bound = zi
+                if np.min(zi_low_bound) < val_min_1d:
+                    val_min_1d = np.min(zi_low_bound)
+                if np.max(zi_high_bound) > val_max_1d:
+                    val_max_1d = np.max(zi_high_bound)
+
+            else:
+                # Lower triangle
+                if not usepartialdependence:
+                    locked_val_list = [
+                        (dim_num, np.array(space.dimensions[dim_num].transform(
+                            [x_vals[dim_num]]), ndmin=2))
+                        for dim_num in range(space.n_dims) if dim_num not in (i, j)]
+                    value_list, stddev_list = generate_random_data(
+                        locked_val_list=locked_val_list,
+                        sample_list=sample_list,
+                        model=model)
+                zi, stddevs = find_average_2D(
+                    value_list,
+                    stddev_list,
+                    included_samples_1d_list=included_list[i],
+                    included_samples_2d_list=included_list[j])
+
+                row.append(
+                    {"xi": xi_list[j], "yi": xi_list[i], "zi": zi, "std": stddevs})
+
+                if np.min(zi) < val_min_2d:
+                    val_min_2d = np.min(zi)
+                if np.max(zi) > val_max_2d:
+                    val_max_2d = np.max(zi)
+                if np.min(stddevs) < stddev_min_2d:
+                    stddev_min_2d = np.min(stddevs)
+                if np.max(stddevs) > stddev_max_2d:
+                    stddev_max_2d = np.max(stddevs)
+
+        plots_data.append(row)
+
+    limits = {
+        # Build all the plots of in the figure
+        "val_1d_min": val_min_1d,
+        "val_1d_max": val_max_1d,
+        "z_min": val_min_2d,
+        "z_max": val_max_2d,
+        "stddev_min": stddev_min_2d,
+        "stddev_max": stddev_max_2d}
+    return plots_data, limits
+
+
+def create_sample_list(space, graph_width, oversampling):
```

Comment:
This function needs a lot more comments and a lot less list comprehension for-loops to be readable. I simply cannot follow what is going on in here.